Mixmatch/streetview_dataset/parse_data_to_tfrecord_main.ipynb

```diff
@@ -0,0 +1,294 @@
+    "\n",
+    "batch_read_write_tfrecords(train_batch_range, INPUT_RECORD_DIR, INPUT_UCF_IMG_DIR, train_writer, detection_property)\n",
+    "batch_read_write_tfrecords(test_batch_range, INPUT_RECORD_DIR, INPUT_UCF_IMG_DIR, test_writer, detection_property)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_writer.close()\n",
+    "test_writer.close()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## streetview_v3\n",
+    "Run this part of code to build streetview_v3 dataset"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Constant input and output path\n",
+    "# NOTE: The images within INPUT_UCF_FILTER_IMG_DIR will be handpicked from outputs of OUTPUT_UCF_IMG_DIR\n",
+    "INPUT_UCF_FILTER_IMG_DIR = './UCF_Streetview_Dataset/test/'\n",
+    "OUTPUT_UCF_IMG_DIR = './UCF_Streetview_Dataset/crop/'\n",
+    "OUTPUT_TEST_RECORD_FILENAME = 'streetview_v3_64-test.tfrecord'\n",
+    "OUTPUT_TRAIN_RECORD_FILENAME = 'streetview_v3_64-train.tfrecord'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### streetview_v3-test\n",
+    "This test set uses all TC11 images as positive cases, and handpick UCF images with confidence < 0.2 as negative cases."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Filter UCF images and copy files for furthur handpicking test set.\n",
+    "res = []\n",
+    "for i in range(int(0.9 * INPUT_RECORD_CNT), INPUT_RECORD_CNT):\n",
+    "    file_name = \"./streetlearn_detections_tfexample-\" + str(i).zfill(5) + \"-of-01000.tfrecord\"\n",
+    "    parsed_image_dataset = read_TFRecord(os.path.join(INPUT_RECORD_DIR, file_name))\n",
+    "    res.extend(filter_image(parsed_image_dataset, INPUT_UCF_IMG_DIR, 0.2))\n",
+    "\n",
+    "for file in res:\n",
+    "    # The format fo the image_name is XXXXXX_Y.jpg, the Y represents the view of the image.\n",
+    "    view = img_name.split('.')[0][-1]\n",
+    "    # ignore sky images\n",
+    "    if view != '5' and view != '0':\n",
+    "        copyfile(INPUT_UCF_IMG_DIR + file, OUTPUT_UCF_IMG_DIR + file)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Build streetview_v3_64-test. Write test tfrecord from TC11 dataset and filtered UCF\n",
+    "test_writer = tf.io.TFRecordWriter(OUTPUT_RECORD_DIR + OUTPUT_TEST_RECORD_FILENAME)\n",
+    "test_image_path = os.path.join(INPUT_TC11_IMG_DIR, 'img')\n",
+    "write_tfrecord_from_images(test_image_path, 1, test_writer)\n",
+    "write_tfrecord_from_images(INPUT_UCF_FILTER_IMG_DIR, 0, test_writer)\n",
```

Comment:
This doesn't look correct to me. It seems you select the entire streetview image instead of the cropped detections as negative examples in test set.

Reply:
Yes, I use the whole image as a negative example in the test set. Because for the whole image, there is no bbox that exceeds the threshold. So I think including the whole image will still be a great example for negative examples.
And also, since the positive examples in the test set is also using whole images from TC11. It would be better for both pos and neg examples are full images in the test set.

Reply:
I was expecting using those low confidence bboxes as negatives. And the whole images in UCF and the whole images in TC11 are different. Images in TC11 are already cropped while images in UCF are raw street view images.

I still suggest using cropped images for consistency across training and testing, although it might not make a big difference in the negative cases.

Reply:
I've changed the filter_image function to make the outputs cropped bboxes. Please take a look at it.